plugins/woocommerce/uninstall.php

```diff
@@ -65,15 +66,17 @@
 	// Delete usermeta.
 	$wpdb->query( "DELETE FROM $wpdb->usermeta WHERE meta_key LIKE 'woocommerce\_%';" );
 
-	// Delete posts + data.
+	// Delete our data from the post and post meta tables, and remove any additional tables we created.
 	$wpdb->query( "DELETE FROM {$wpdb->posts} WHERE post_type IN ( 'product', 'product_variation', 'shop_coupon', 'shop_order', 'shop_order_refund' );" );
 	$wpdb->query( "DELETE meta FROM {$wpdb->postmeta} meta LEFT JOIN {$wpdb->posts} posts ON posts.ID = meta.post_id WHERE posts.ID IS NULL;" );
 
 	$wpdb->query( "DELETE FROM {$wpdb->comments} WHERE comment_type IN ( 'order_note' );" );
 	$wpdb->query( "DELETE meta FROM {$wpdb->commentmeta} meta LEFT JOIN {$wpdb->comments} comments ON comments.comment_ID = meta.comment_id WHERE comments.comment_ID IS NULL;" );
 
-	$wpdb->query( "DROP TABLE IF EXISTS {$wpdb->prefix}woocommerce_order_items" );
-	$wpdb->query( "DROP TABLE IF EXISTS {$wpdb->prefix}woocommerce_order_itemmeta" );
+	foreach ( wc_get_container()->get( OrdersTableDataStore::class )->get_all_table_names() as $cot_table ) {
```

Comment:
`woocommerce_order_items` and `woocommerce_order_itemmeta` are different tables from COT and needs to be deleted (if we are deleting tables).

Reply:
> If we are deleting tables

Well ... we *are* and have been deleting tables during uninstall. So, this change is mostly for consistency (otherwise, I envision a future complaint where someone runs the uninstall process, and everything is cleaned up except for this subset of our tables).

> woocommerce_order_items and woocommerce_order_itemmeta are different tables from COT and needs to be deleted

To be clear, I removed lines 75-76 because they are superfluous and not because I wanted us to keep those tables. Up on line 60 you will see a call to [`WC_Install::remove_tables()`](https://github.com/woocommerce/woocommerce/blob/ce3ce7cb869e9a50419b32bf60f6dd6d53195210/plugins/woocommerce/uninstall.php#L60) which was already taking care of this (in my testing this was perfectly fine, but let me know if you found a case where those two tables still manage to survive).